ecdsa/keygen/dln_verifier.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2019 Binance
+//
+// This file is part of Binance. The full Binance copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+package keygen
+
+import (
+	"errors"
+	"math/big"
+	"runtime"
+)
+
+type DlnProofVerifier struct {
+	semaphore chan interface{}
+}
+
+func NewDlnProofVerifier(optionalConcurrency ...int) *DlnProofVerifier {
```

Comment:
The next step will be to find a way to configure/enhance the verifier. The best option I can think of now is:
1. Introduce `KeyGenerationParameters` similar to `ReSharingParameters`.
2. Allow replacing the default DLN verifier with a custom one. This will be an expert-mode function, setting DLN verifier will not be available from `NewParameters` / `NewKeyGenerationParameters`.

Also, worth noting the same problem with DLN verification exists for re-sharing protocol. This will also be fixed as a part of the work on this PR.